tests/unittests/storage_tests.py

```
@@ -0,0 +1,1097 @@
+    def test_ensure_dir(self, temp_dir):
+        path = os.path.join(temp_dir, "new_dir")
+        fos.ensure_dir(path)
+        assert os.path.isdir(path)
+
+    def test_ensure_empty_dir_raises_if_exists(self, temp_dir):
+        dir_path = os.path.join(temp_dir, "existing_dir")
+        os.makedirs(dir_path)
+        with open(os.path.join(dir_path, "existing.txt"), "w") as f:
+            f.write("existing")
+
+        with pytest.raises(ValueError):
```

Ruff (0.14.7)

`pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)